ozaria/site/components/avatar-selector/PageAvatarSelector/index.vue

```diff
@@ -0,0 +1,216 @@
+
+<script>
+import LayoutCenterContent from '../../common/LayoutCenterContent.vue'
+import LayoutAspectRatioContainer from '../../common/LayoutAspectRatioContainer.vue'
+import LayoutChrome from '../../common/LayoutChrome.vue'
+import Surface from '../../char-customization/common/Surface.vue';
+import BaseButton from '../../common/BaseButton.vue';
+
+// TODO migrate api calls to the Vuex store.
+import { getThangTypeOriginal } from '../../../../../app/core/api/thang-types'
+const ThangType = require('models/ThangType')
+
+// TODO replace placeholders with the various options.
+const avatarCinematicOriginalId = {
+  star: '5c373c9f9034ac0034b43b22'
+}
+
+export default Vue.extend({
+  components: {
+    LayoutChrome,
+    LayoutCenterContent,
+    LayoutAspectRatioContainer,
+    BaseButton,
+    Surface
+  },
+  
+  data: () => ({
+    selected: -1,
+    loadedThangTypes: {},
+    loaded: false
+  }),
+
+  async created () {
+    if (!me.hasAvatarSelectorAccess())  {
+      return application.router.navigate('/', { trigger: true })
+    }
+    const loader = []
+
+    for (const avatarKey of Object.keys(avatarCinematicOriginalId)) {
+      const thangLoading = getThangTypeOriginal(avatarCinematicOriginalId[avatarKey])
```

Comment:
Is it more efficient to use promises here, rather than `await`'ing? Unless I'm mistaken, it would behave the same way if you `await`ed for each of these in the loop, rather than waiting for all the promises to finish?

Reply:
I believe if you awaited each one in the loop, it wouldn't kick off the next request until the prior await had completed. Thus each request would be initiated and resolved sequentially.

The way it is currently written, all the requests start, and then we await for them all to finish concurrently.